### PR TITLE
Cleanup DPS service enrollment samples and SDK code

### PR DESCRIPTION
**Pull request by drwill-ms**
Similar to #3070

### CODE REVIEW

File: azureiot.sln [hunk 1]
```diff
@@ -152,11 +152,9 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SymmetricKeySample", "provi
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "InvokeDeviceMethod", "iothub\service\samples\getting started\InvokeDeviceMethod\InvokeDeviceMethod.csproj", "{A9058CEC-E0EB-4461-A1B0-4C6CB83A8AAC}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "EnrollmentGroupX509Sample", "provisioning\service\samples\getting started\EnrollmentGroupX509Sample\EnrollmentGroupX509Sample.csproj", "{83170B36-375E-40A9-B25D-999072362626}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "EnrollmentGroupSample", "provisioning\service\samples\getting started\EnrollmentGroupSample\EnrollmentGroupSample.csproj", "{83170B36-375E-40A9-B25D-999072362626}"
```
Comment (drwill-ms) [inline azureiot.sln:155]: Changed X509 to symmetric key because the type of authentication for this sample seems irrelevant, and this simplifies execution.

File: azureiot.sln [hunk 2]
```diff
@@ -152,11 +152,9 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SymmetricKeySample", "provi
...
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "IndividualEnrollmentX509Sample", "provisioning\service\samples\getting started\IndividualEnrollmentX509Sample\IndividualEnrollmentX509Sample.csproj", "{8F8EF9CB-EF6A-4E39-9CE1-8CD06C1AB255}"
-EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "IndividualEnrollmentTpmSample", "provisioning\service\samples\getting started\IndividualEnrollmentTpmSample\IndividualEnrollmentTpmSample.csproj", "{705C5B5A-6166-4566-B167-4401491D7650}"
```
Comment (drwill-ms) [inline azureiot.sln:159]: Removed TPM as it is mostly a dupe of the other one.

**Review by patilsnr — APPROVED** [review]

File: provisioning/service/samples/getting started/EnrollmentGroupSample/EnrollmentGroupSample.cs [hunk 3]
```diff
@@ -3,81 +3,87 @@
...
-            await QueryEnrollmentGroupAsync().ConfigureAwait(false);
             await CreateEnrollmentGroupAsync().ConfigureAwait(false);
             await GetEnrollmentGroupInfoAsync().ConfigureAwait(false);
+            await QueryEnrollmentGroupAsync().ConfigureAwait(false);
             await DeleteEnrollmentGroupAsync().ConfigureAwait(false);
         }
 
+        public async Task CreateEnrollmentGroupAsync()
+        {
+            Attestation attestation = new SymmetricKeyAttestation();
```
Comment (abhipsaMisra) [inline provisioning/service/samples/getting started/EnrollmentGroupSample/EnrollmentGroupSample.cs:31]: 👍🏽

File: provisioning/service/src/Models/SymmetricKeyAttestation.cs [hunk 4]
```diff
@@ -11,11 +11,16 @@ namespace Microsoft.Azure.Devices.Provisioning.Service
...
-        [JsonConstructor]
+        public SymmetricKeyAttestation()
+        { }
+
+        /// <summary>
+        /// Creates an instance of this class with the specified keys.
+        /// </summary>
+        /// <param name="primaryKey">The primary key to use for attestation.</param>
+        /// <param name="secondaryKey">The secondary key to use for attestation.</param>
         public SymmetricKeyAttestation(string primaryKey, string secondaryKey)
```
Comment (abhipsaMisra) [inline provisioning/service/src/Models/SymmetricKeyAttestation.cs:24]: Q - should we update this to have secondaryKey default to null?

**Review by abhipsaMisra — APPROVED** [review]

**Review by schoims — APPROVED** [review]

**Review by andyk-ms — APPROVED** [review]

**Review by timtay-microsoft — APPROVED** [review]
